modules/jdbc/src/main/java/org/apache/ignite/internal/jdbc/JdbcConnection.java

```diff
@@ -367,6 +374,20 @@ public void rollback(Savepoint savepoint) throws SQLException {
         throw new SQLFeatureNotSupportedException("Savepoints are not supported.");
     }
 
+    /**
+     * Finish transaction.
+     *
+     * @param commit {@code True} to commit, {@code false} to rollback.
+     * @throws SQLException If failed.
+     */
+    private void finishTx(boolean commit) throws SQLException {
+        JdbcFinishTxResult res = handler().finishTxAsync(connectionId, commit).join();
```

Comment:
join() is uninterruptible call.

Reply:
I mean the thread waiting on future.get() can be interrupted and a client application can continue work.
The future.join() ignores interrupts and the thread will hang forever. The only way is to kill JVM.
We have faced similar issues with future.join() on TC that lead to test suite hanging.

Reply:
I understood that you are suggesting to replace `join()` with `get()`, and I indicated above why I am not sure that it is worth doing this specifically for commit() and rollback().
As I see our jdbc driver implementation currently does not handle thread interruptions at all (except client connect).

Are you suggesting to change all join() calls to get() there and wrap InterruptedException in SqlException?

And as I pointed above, the jdbc specification does not describe interruptionexception handling. The operation is expected to be aborted via JdbcConnection#cancel.

Reply:
@AMashenkov  Thanks, finally I did replace all ``join()` to `get()`.
Since thread interruption is not described in any way by jdbc, then the user should not use it, but it can really help us in development and testing.
The only note - at the moment I do not restore the thread interruption flag when handling `InterruprionException`, please see the changes in the last two commits, is that ok?